src/css-select-parse5-adapter.ts

```diff
@@ -0,0 +1,94 @@
+import {Adapter as CSSSelectAdapter} from 'css-select';
+import {Attribute, Element, Node, TreeAdapter} from 'parse5';
+const defaultTreeAdapter = require('parse5/lib/tree-adapters/default');
+export type Predicate = (node: Node) => boolean;
+export class Adapter implements CSSSelectAdapter<Node, Element> {
+  treeAdapter: TreeAdapter;
+  constructor(treeAdapter: TreeAdapter) {
+    this.treeAdapter = treeAdapter;
+  }
+  isTag(node: Node): node is Element {
+    return this.treeAdapter.isElementNode(node);
+  }
+  existsOne(test: Predicate, elems: Element[]): boolean {
+    return elems.some(test);
+  }
+  getAncestors(node: Node): Node[] {
+    const parentNode = this.hasParent(node) && this.getParent(node);
+    return parentNode ? [parentNode, ...this.getAncestors(parentNode)] : [];
```

Comment:
For recursive list building, I like to pass the list to add to instead of constructing new ones at each level:

```ts
getAncestors(node: Node): Node[] {
  return this._getAncestors(node, []);
}

_getAncestors(node: Node, ancestors: Node[]): Node[] {
  const parent = this.getParent(node);
  if (parent !== undefined) {
    ancestors.push(parent)
    this._getAncestors(parent, ancestors);
  }
  return ancestors;
}
```

Reply:
That's a sensible refactor.  I was being lazy.  Upon further reflection recursion not needed anyways.  New version:

```ts
  getAncestors(node: Node): Node[] {
    const ancestors: Node[] = [];
    do {
      node = this.getParent(node);
      if (node) {
        ancestors.push(node);
      }
    } while (node);
    return ancestors;
  }
```